### PR TITLE
Work in progress #23

### PR DESCRIPTION
**Pull request by captain0612**
Covered service layer as well as repository layer for unit test

### CODE REVIEW

File: src/test/java/org/erp/school/document/service/impl/DocumentServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        documentRepository = mock(DocumentRepository.class);
+        clientRepository = mock(ClientRepository.class);
+        documentService = new DocumentServiceImpl(documentRepository,clientRepository);
+    }
+
+    @Test
+    void fetchDocumentsClientId() {
+        when(clientRepository.findById(eq("Test_Client"))).thenReturn(ClientPrototype.getTestClient());
+        List<Document> testDocs =  documentService.fetchDocumentsClientId("Test_Client");
+        System.out.println(testDocs);
```
Comment (patricksan) [inline src/test/java/org/erp/school/document/service/impl/DocumentServiceImplTest.java]: use `log.info`.
There is a lombok for that: 

```
@Slf4j
public class LogExampleOther {
  
  public static void main(String... args) {
    log.error("Something else is wrong here");
  }
}
```

**Review by patricksan — CHANGES_REQUESTED** [review]

**Comment by patricksan** [comment]
We could also have an integration test. That would start to test from the Controller.
After we commit these classes I can do that.

**Review by Aurrix — APPROVED** [review]

**Comment by patricksan** [comment]
Hey @Aurrix , I did some comments in the PR, but it seems you merged with @captain0612 fix them. :-)

**Comment by captain0612** [comment]
yea @patricksan we were trying to merge my changes and @Aurrix changes to resolve conflicts, anyway ill be be rising one more PR with the changes again :p

**Pull request merged**